internal/internal_decision_state_machine.go

```diff
@@ -917,22 +919,26 @@ func (h *decisionsHelper) handleChildWorkflowExecutionCanceled(workflowID string
 
 func (h *decisionsHelper) getDecisions(markAsSent bool) []*s.Decision {
 	var result []*s.Decision
-	for _, d := range h.orderedDecisions {
+	for curr := h.orderedDecisions.Front(); curr != nil; {
+		next := curr.Next() // get next item here as we might need to remove curr in the loop
+		d := curr.Value.(decisionStateMachine)
 		decision := d.getDecision()
 		if decision != nil {
 			result = append(result, decision)
 		}
-	}
 
-	if markAsSent {
-		h.handleDecisionsSent()
-	}
+		if markAsSent {
+			d.handleDecisionSent()
```

Comment:
This seems like a change in behavior? Earlier all decisions the orderedDecisions were updated. Why is that?

Reply:
there is no behavior change. Previously it was done in 2 for loops (second one done in separate function), now they are done in one loop.

Reply:
@madhuravi , previously it is done out of the first for loop.